store/memory.go

```diff
@@ -86,44 +100,110 @@ func (e *MemStore) pruneExpired() {
 }
 
 // Get fetches a value from the store.
-func (e *MemStore) Get(ctx context.Context, commit []byte) ([]byte, error) {
+func (e *MemStore) Get(ctx context.Context, commit []byte, domain eigendacommon.DomainType) ([]byte, error) {
 	e.RLock()
 	defer e.RUnlock()
 
-	key := common.Bytes2Hex(commit)
-	if _, exists := e.store[key]; !exists {
+	var cert *disperser.BlobInfo
+	err := rlp.DecodeBytes(commit, &cert)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode DA cert to RLP format: %w", err)
+	}
+
+	var encodedBlob []byte
+	var exists bool
+	if encodedBlob, exists = e.store[string(commit)]; !exists {
 		return nil, fmt.Errorf("commitment key not found")
 	}
 
-	return e.store[key], nil
+	// Don't need to do this really since it's a mock store
+	err = e.verifier.Verify(cert.BlobHeader.Commitment, encodedBlob)
+	if err != nil {
+		return nil, err
+	}
+
+	switch domain {
+	case eigendacommon.BinaryDomain:
+		return e.codec.DecodeBlob(encodedBlob)
+	case eigendacommon.PolyDomain:
+		return encodedBlob, nil
+	default:
+		return nil, fmt.Errorf("unexpected domain type: %d", domain)
+	}
 }
 
 // Put inserts a value into the store.
 func (e *MemStore) Put(ctx context.Context, value []byte) ([]byte, error) {
+	if uint64(len(value)) > e.maxBlobSizeBytes {
+		return nil, fmt.Errorf("blob is larger than max blob size: blob length %d, max blob size %d", len(value), e.maxBlobSizeBytes)
+	}
+
 	e.Lock()
 	defer e.Unlock()
 
-	fingerprint := crypto.Keccak256Hash(value)
-	// add some entropy to commit to emulate randomness seen in EigenDA
-	// when generating operator BLS signature certificates
+	encodedVal, err := e.codec.EncodeBlob(value)
+	if err != nil {
+		return nil, err
+	}
+
+	commitment, err := e.verifier.Commit(encodedVal)
+	if err != nil {
+		return nil, err
+	}
+
+	// generate batch header hash
 	entropy := make([]byte, 10)
-	_, err := rand.Read(entropy)
+	_, err = rand.Read(entropy)
 	if err != nil {
 		return nil, err
 	}
+	mockBatchHeaderHash := crypto.Keccak256Hash(entropy)
+
+	// only filling out commitment fields for now
+	cert := &disperser.BlobInfo{
+		BlobHeader: &disperser.BlobHeader{
+			Commitment: &grpccommon.G1Commitment{
+				X: commitment.X.Marshal(),
+				Y: commitment.Y.Marshal(),
+			},
+			// DataLength: ,
+			// BlobQuorumParams: ,
+		},
+		BlobVerificationProof: &disperser.BlobVerificationProof{
+			BatchMetadata: &disperser.BatchMetadata{
+				BatchHeader: &disperser.BatchHeader{
+					// BatchRoot: ,
+					// QuorumNumbers: ,
+					// QuorumSignedPercentages: ,
+					// ReferenceBlockNumber: ,
+				},
+				// SignatoryRecordHash: ,
+				// Fee: ,
+				// ConfirmationBlockNumber: ,
+				BatchHeaderHash: mockBatchHeaderHash[:],
+			},
+			// BatchId: ,
+			// BlobIndex: ,
+			// InclusionProof: ,
```

Comment:
think we'll need to fill these randomly as well since they're extracted sequencer side for encoding into the certificate thats sent on-chain and verified against the service manager

Reply:
Is there any point to filling this randomly if they're not going to validate onchain? Seems like empty values about as good in that case.